application/teams/team.py

```diff
@@ -1,14 +1,15 @@
 """Access to Team objects in Ne4J"""
+from uuid import uuid4
 from neomodel import StructuredNode, StringProperty, RelationshipTo, cardinality
 
 from application.model_mixins import DefaultPropertyMixin, DefaultHelperMixin
 from application.teams.team_schema import TeamSchema
 
-
 class Team(StructuredNode, DefaultPropertyMixin, DefaultHelperMixin):  # pylint:disable=abstract-method
     """The class that manages Teams"""
     schema = TeamSchema
     name = StringProperty(required=True)
+    join_key = StringProperty(required=False, default=uuid4)
```

Comment:
We can make the property unique here and add an index (`unique_index=true`). That way team creation will fail if by any chance the generated key already exists.